zad2/warriors.rb

```diff
@@ -1,12 +1,72 @@
-class Warrior
+class Character
+  attr_reader :name
+  attr_accessor :level
+
   def initialize(name:, level:)
+    @name = name
+    @level = level.between?(1, 99) ? level : 1
   end
 
   def strength
+    @level + [*1..12].sample
+  end
+
+  def card
+    "#{@name} (lvl #{@level})"
   end
 end
 
+
+class Warrior < Character
+end
+
+class Monster < Character
+end
+
 class BattleArena
+  attr_accessor :first_character, :second_character
+
   def initialize(first_character, second_character)
+    @first_character = first_character
+    @second_character = second_character
+    @first_character_dmg = @first_character.strength
+    @second_character_dmg = @second_character.strength
+  end
+
+  def battle!
+    puts "#{@first_character.card} attacked #{@second_character.card} with #{@first_character_dmg} damage"
+    puts "#{@second_character.card} attacked #{@first_character.card} with #{@second_character_dmg} damage"
+    compare_dmg
+  end
+
+  private
+  
+  def compare_dmg
+    if @first_character_dmg === @second_character_dmg
```

Comment:
co do `===` ruby to nie JavaScript ;)
Jeśli porównujesz dwa Integery, `==` to wszystko co potrzebujesz

Reply:
`===` raczej bardzo rzadko się używa.

najczęściej jest on używany w `case` "po cichu". Dzięki temu można robić konstrukcje typu:
```
case a
when 'some string' then x
when Integer then y
end
```

Reply:
Dzięki :) Machnąłem tak z przyzwyczajenia.